src/modules/main/PlayerView.tsx

```diff
@@ -1,23 +1,169 @@
-import { Box, Typography } from '@mui/material';
+import { useEffect, useMemo, useState } from 'react';
 
-import { useLocalContext } from '@graasp/apps-query-client';
+import {
+  Alert,
+  Box,
+  Button,
+  CircularProgress,
+  Container,
+  Stack,
+  Typography,
+} from '@mui/material';
 
-import { hooks } from '@/config/queryClient';
+import isEqual from 'lodash.isequal';
+
+import { AnsweredLabel, Label, Settings, SettingsKeys } from '@/@types';
+import { ANSWERS_SUBMISSION_TYPE } from '@/config/constants';
+import { useAppTranslation } from '@/config/i18n';
+import { hooks, mutations } from '@/config/queryClient';
 import { PLAYER_VIEW_CY } from '@/config/selectors';
+import { APP } from '@/langs/constants';
+
+import PlayerFrame from '../common/PlayerFrame';
+
+type SubmittedAnswer = {
+  expectedId: string;
+  actualId?: string;
+};
 
 const PlayerView = (): JSX.Element => {
-  const { permission } = useLocalContext();
   const { data: appContext } = hooks.useAppContext();
-  const members = appContext?.members;
+  const { t } = useAppTranslation();
+  const { data: appData } = hooks.useAppData<{ answers: SubmittedAnswer[] }>();
+  const { data: appSettings, isLoading } = hooks.useAppSettings<Settings>({
+    name: SettingsKeys.SettingsData,
+  });
+  const { mutate: saveAppData } = mutations.usePostAppData();
+  const [answeredLabels, setAnsweredLabels] = useState<AnsweredLabel[]>([]);
+  // labels will be null only before setting the state as we cannot render all labels within container if not settled yet
+  const [labels, setLabels] = useState<null | Label[]>(null);
+
+  const { data: image } = hooks.useAppSettings({
+    name: SettingsKeys.File,
+  });
+
+  const answersAppData = appData?.filter(
+    ({ type }) => type === ANSWERS_SUBMISSION_TYPE,
+  );
+  const settingLabels = appSettings?.[0].data.labels;
+  const lastAnswerAppData = answersAppData?.[answersAppData.length - 1];
```

Comment:
I think it's safer if you actually take the last created/updated app data instead of the last one in the array.